rebasehelper/checkers/abipkgdiff_tool.py

```diff
@@ -112,17 +119,80 @@ def run_check(cls, results_dir, **kwargs):
             command.append(pkg)
             command.append(new_pkg)
             logger.debug('Package name for ABI comparison %s', old_name)
-            output = os.path.join(cls.results_dir, results_dir, old_name + '-' + cls.log_name)
+            output = os.path.join(cls.results_dir, old_name + '.txt')
             try:
                 ret_code = ProcessHelper.run_subprocess(command, output_file=output)
             except OSError:
                 raise CheckerNotFoundError("Checker '%s' was not found or installed." % cls.CMD)
 
             if int(ret_code) & settings.ABIDIFF_ERROR and int(ret_code) & settings.ABIDIFF_USAGE_ERROR:
                 raise RebaseHelperError('Execution of %s failed.\nCommand line is: %s' % (cls.CMD, cmd))
-            if int(ret_code) == 0:
-                text = 'ABI of the compared binaries in package %s are equal.' % old_name
-            else:
-                text = 'ABI of the compared binaries in package %s are not equal.' % old_name
-            reports[output] = text
-        return reports
+            reports[old_name] = int(ret_code)
+        return {'packages': cls.parse_abi_logs(reports), 'abi_changes': cls.abi_changes, 'path': cls.results_dir}
+
+    @classmethod
+    def parse_abi_logs(cls, reports):
+        """
+        Parse abipkgdiff logs
+        :param reports: dictionary with paths to the logs of the produced (sub)packages
+        :return: returns packages names with its abipkgdiff changes
+        """
+        def parse_changes(chgs):
+            """
+            Parses each line with abipkgdiff output info
+            :param chgs: lines of changes
+            :return: dictionary of list of changes
+            """
+            ret_changes = {}
+            for line in chgs:
+                # Find all numbers of added/changed/deleted and get rid of '0' changes
+                m = re.search(r'(^[\w\s]+:)'
+                              r'.{1}([\d]+ Removed[\s\w\d()]*)'
+                              r'.{2}([\d]+ Changed[\s\w\d()]*)'
+                              r'.{2}([\d]+ Added[\s\w\d()]*)',
+                              line)
+                if not m:
+                    continue
+                # If any useful info found, store it
+                ret_changes[m.group(1)] = [change for change in m.groups()[1:]]
+            return ret_changes
+
+        pkgs = {}
+        for pkg, ret_code in six.iteritems(reports):
+            cur_pkg = {}
+            if ret_code:
+                with open(os.path.join(cls.results_dir, pkg + '.txt'), 'r') as f:
+                    lines = f.readlines()
```

Comment:
How is this better or more readable than for example this?

```python
def parse(lines):
    summary_re = re.compile(r'''^
        (?P<kind>\w+\s+changes\s+summary):\s+
        (?P<removed>\d+)\s+Removed(\s+\((?P<removed_filtered_out>\d+)\s+filtered\s+out\))?,\s+
        (?P<changed>\d+)\s+Changed(\s+\((?P<changed_filtered_out>\d+)\s+filtered\s+out\))?,\s+
        (?P<added>\d+)\s+Added(\s+\((?P<added_filtered_out>\d+)\s+filtered\s+out\))?\s+
        .*$
        ''', re.VERBOSE)
    result = {}
    for line in lines:
        m = summary_re.match(line)
        if m:
            d = m.groupdict()
            result[d['kind']] = d
    return result

...

    changes = parse(f.readlines())
```

Of course, this expects that order of `Removed`, `Changed`, `Added` is fixed (just like your implementation). More robust solution would be this:

```python
def parse(lines):
    summary_re = re.compile(r'''^
        (?P<kind>\w+\s+changes\s+summary):\s+
        (?P<changes>.+)
        $
        ''', re.VERBOSE)
    change_re = re.compile(r'''
        (?P<count>\d+)\s+
        (?P<what>Added|Changed|Removed)
        (\s+\((?P<filtered_out>\d+)\s+filtered\s+out\))?
        ''', re.VERBOSE)
    result = {}
    for line in lines:
        ms = summary_re.match(line)
        if ms:
            ds = ms.groupdict()
            result[ds['kind']] = {}
            for mc in changes_re.finditer(ds['changes']):
                dc = mc.groupdict()
                result[ds['kind']][dc['what']] = dc
    return result

...

    changes = parse(f.readlines())
```

Reply:
Thanks, this is way more readable and maintainable!